node/src/reactor/validator.rs

```diff
@@ -874,10 +874,11 @@ impl reactor::Reactor for Reactor {
                         effects.extend(self.dispatch_event(effect_builder, rng, reactor_event));
                         effects
                     }
-                    ConsensusAnnouncement::Handled(_) => {
-                        debug!("Ignoring `Handled` announcement in `validator` reactor.");
-                        Effects::new()
-                    }
+                    ConsensusAnnouncement::CreatedFinalitySignature(fs) => self.dispatch_event(
+                        effect_builder,
+                        rng,
+                        Event::LinearChain(linear_chain::Event::FinalitySignatureReceived(fs)),
+                    ),
```

Comment:
In both `joiner` and `validator` reactors this announcement is routed to the `LinearChain`. Previously it was a request-event and I still think it should be. We should not assume implicitly that a `BlockAdded` announcement will trigger a consensus component to create a finality signature that will be routed back to the `LinearChain`.

Reply:
And in the future there will be no Era Supervisor in the Joiner, so if we made this a request, we would effectively be working against ourselves. The point of this PR is to make the Era Supervisor removable from the Joiner, so this _can't_ be a request.

Reply:
You're right – then we can't make it a request.